internal/devbox/session_manager.go

```diff
@@ -0,0 +1,453 @@
+package devbox
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"log/slog"
+
+	"github.com/signadot/cli/internal/auth"
+	"github.com/signadot/cli/internal/buildinfo"
+	"github.com/signadot/cli/internal/config"
+	"github.com/signadot/go-sdk/client"
+	sdkauth "github.com/signadot/go-sdk/client/auth"
+	"github.com/signadot/go-sdk/client/devboxes"
+	"github.com/signadot/go-sdk/transport"
+	"github.com/spf13/viper"
+)
+
+const (
+	// RenewalInterval is how often to renew the devbox session
+	RenewalInterval = 1 * time.Minute
+	// RenewalJitter adds randomness to avoid thundering herd
+	RenewalJitter = 30 * time.Second
+)
+
+type SessionManager struct {
+	log            *slog.Logger
+	ciConfig       *config.ConnectInvocationConfig
+	apiClient      *client.SignadotAPI
+	renewalTicker  *time.Ticker
+	doneCh         chan struct{}
+	shutdownCh     chan struct{}
+	sessionReleased bool
+	lastError      error
+	lastErrorTime  time.Time
+	mu             sync.RWMutex
+}
+
+func NewSessionManager(log *slog.Logger, ciConfig *config.ConnectInvocationConfig, shutdownCh chan struct{}) (*SessionManager, error) {
+	if ciConfig.DevboxID == "" || ciConfig.DevboxSessionID == "" {
+		// No devbox session to manage
+		return nil, nil
+	}
+
+	// Resolve auth dynamically
+	authInfo, err := auth.ResolveAuth()
+	if err != nil {
+		return nil, fmt.Errorf("failed to resolve auth: %w", err)
+	}
+	if authInfo == nil {
+		return nil, fmt.Errorf("no auth found")
+	}
+
+	// Create API client with dynamic auth resolution
+	apiClient, err := createAPIClient(ciConfig, authInfo)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create API client: %w", err)
+	}
+
+	dsm := &SessionManager{
+		log:        log.With("component", "devbox-session-manager"),
+		ciConfig:   ciConfig,
+		apiClient:  apiClient,
+		doneCh:     make(chan struct{}),
+		shutdownCh: shutdownCh,
+	}
+
+	return dsm, nil
+}
+
+func createAPIClient(ciConfig *config.ConnectInvocationConfig, authInfo *auth.ResolvedAuth) (*client.SignadotAPI, error) {
```

Comment:
Why can't we use the normal API client and we need a custom one? (note that we do auth refresh in the regular one)